src/cargo/ops/cargo_rustc/context.rs

```diff
@@ -803,8 +803,9 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
     }
 
     pub fn show_warnings(&self, pkg: &PackageId) -> bool {
-        pkg == &self.current_package || pkg.source_id().is_path() ||
-            self.config.extra_verbose()
+        self.ws.current_opt().map_or(false, |p| *pkg == *p.package_id())
+            || pkg.source_id().is_path()
+            || self.config.extra_verbose()
```

Comment:
Well technically actually, we should show warnings for all *workspace members*, not not all path dependencies.